src/client/scripts/esm/game/rendering/draganimation.js

```diff
@@ -0,0 +1,131 @@
+
+// Import start
+import shapes from "./shapes.js";
+import buffermodel from "./buffermodel.js";
+import bufferdata from "./bufferdata.js";
+import options from "./options.js";
+import spritesheet from "./spritesheet.js";
+import perspective from "./perspective.js";
+import sound from "../misc/sound.js";
+import frametracker from "./frametracker.js";
+import movement from "./movement.js";
+import input from "../input.js";
+// Import end
+
+"use strict";
+
+/**
+ * This script hides the original piece and renders a copy at the pointer location.
+ * It also highlights the square that the piece would be dropped on (to do)
+ * and plays the sound when the piece is dropped.
+ */
+
+/**
+ * To Do:
+ * - Emphasise the hovered square like Lichess and Chess.com.
+ */
+
+const z = 0.01;
+/** When not in perspective the pieces size is independent of board scale. */
+const touchscreenScale = 2;
+const mouseScale = 1;
+/** When using a touchscreen, the piece is shifted upward by this amount to prevent it being covered by fingers. */
+const touchscreenOffset = 2;
+/**
+ * The minimum size of the dragged piece relative to the stationary pieces.
+ * When zoomed in, this prevents it becoming tiny relative to the others.
+ */
+const minimumScale = 0.75;
+
+/** The hight the piece is rendered above the board when in perspective mode. */
+const perspectiveHeight = 0.6;
+const shadowColor = [0.1, 0.1, 0.1, 0.5];
+
+let startCoords;
+let endCoords;
+let pieceType;
+
+let transparentModel;
+let pieceModel;
```

Comment:
Since the models are generated and rendered in the rendering methods, they don't need to be in the public scope of the script.